tests/api_fastapi/core_api/routes/public/test_config.py

```diff
@@ -0,0 +1,317 @@
+
+                    [{SECTION_DATABASE}]
+                    {OPTION_KEY_SQL_ALCHEMY_CONN} = {OPTION_VALUE_SQL_ALCHEMY_CONN}
+                    """
+                ),
+            ),
+            (
+                None,
+                HEADERS_INVALID,
+                422,
+                {
+                    "detail": [
+                        {
+                            "type": "literal_error",
+                            "loc": ["header", "content-type"],
+                            "msg": "Input should be 'application/json' or 'text/plain'",
+                            "input": "invalid",
+                            "ctx": {"expected": "'application/json' or 'text/plain'"},
+                        }
+                    ]
+                },
+            ),
+            (
+                SECTION_CORE,
+                HEADERS_JSON,
+                200,
+                {
+                    "sections": [
+                        {
+                            "name": SECTION_CORE,
+                            "options": [
+                                {"key": OPTION_KEY_PARALLELISM, "value": OPTION_VALUE_PARALLELISM},
+                            ],
+                        },
+                    ],
+                },
+            ),
+            (
+                SECTION_SMTP,
+                HEADERS_TEXT,
+                200,
+                textwrap.dedent(
+                    f"""\
+                    [{SECTION_SMTP}]
+                    {OPTION_KEY_SMTP_HOST} = {OPTION_VALUE_SMTP_HOST}
+                    {OPTION_KEY_SMTP_MAIL_FROM} = {OPTION_VALUE_SMTP_MAIL_FROM}
+                    """
+                ),
+            ),
+            (
+                SECTION_DATABASE,
+                HEADERS_JSON,
+                200,
+                {
+                    "sections": [
+                        {
+                            "name": SECTION_DATABASE,
+                            "options": [
+                                {"key": OPTION_KEY_SQL_ALCHEMY_CONN, "value": OPTION_VALUE_SQL_ALCHEMY_CONN},
+                            ],
+                        },
+                    ],
+                },
+            ),
+            (None, HEADERS_JSON, 403, FORBIDDEN_RESPONSE),
+            (SECTION_CORE, HEADERS_JSON, 403, FORBIDDEN_RESPONSE),
+            (SECTION_NOT_EXIST, HEADERS_JSON, 404, {"detail": f"Section {SECTION_NOT_EXIST} not found."}),
+        ],
+    )
+    def test_get_config(self, test_client, section, headers, expected_status_code, expected_response):
+        query_params = {"section": section} if section else None
+        if expected_status_code == 403:
+            with conf_vars(AIRFLOW_CONFIG_DISABLE_EXPOSE_CONFIG):
+                response = test_client.get("/public/config/", headers=headers, params=query_params)
+        else:
+            response = test_client.get("/public/config/", headers=headers, params=query_params)
```

Comment:
I think we are missing a test with `non-sensitive-only` ?

Reply:
It seems the legacy API missed the non-sensitive-only case in the get_config endpoint. I’ll add this case along with a test. Thanks !

Reply:
I think you can take inspiration from `test_should_respond_200_text_plain_with_non_sensitive_only` from legacy tests.